LRA coordinator feature

idlewis opened the pull request:
Add the new coordinator feature

tevans78 commented on dev/io.openliberty.microprofile.lra.coordinator.1.0.internal/resources/OSGI-INF/wlp/defaultInstances.xml:
```diff
@@ -0,0 +1,13 @@
+        IBM Corporation - initial implementation
+ -->
+<server>
+    <application context-root="/lrac" type="war" id="lra-c" location="${wlp.install.dir}/lib/mpLRACoordinator_5.10.5.jar"/>
```
is `/larc` a standard context root? I presume that it can be customized by overriding this line in the server.xml?

idlewis replied:
It was a random choice by me. I have an issue open to add config for this. I'll add a comment in the issue to say that a sensible default also needs to be chosen

tevans78 replied:
I suspect that is the kind of detail that should have been in the UFO? If it wasn't please update the UFO.

tevans78 commented on dev/io.openliberty.org.jboss.narayana.rts.lra-coordinator/bnd.bnd:
```diff
@@ -0,0 +1,61 @@
+# is made to support more of the optional parts of narayana in future, 
+# some of the excluded packages may need to be added back 
+Import-Package: \
+    !android.os, \
+    !com.arjuna.ats.internal.jta, \
+    !com.sun.tools.jconsole, \
+    !org.apache.activemq.artemis.core*, \
+    !org.jfree*, \
+    !org.eclipse.microprofile.openapi*, \
```
Interesting list of packages to exclude? How did you arrive at this list?

idlewis replied:
Running the build and looking for bnd import failures, and investigation of what they are used for.
They are I believe all for optional parts of narayana which are not needed by the LRA implementation

tevans78 commented on dev/cnf/oss_dependencies.maven:
```diff
@@ -432,11 +432,18 @@ org.jboss.arquillian.core:arquillian-core-spi:1.3.0.Final
 org.jboss.classfilewriter:jboss-classfilewriter:1.1.2.Final
 org.jboss.logging:jboss-logging:3.4.1.Final
+org.jboss.narayana.arjunacore:arjuna:5.10.5.Final
 org.jboss.narayana.rts:lra-client:5.10.5.Final
+org.jboss.narayana.rts:lra-client:5.10.5.Final
+org.jboss.narayana.rts:lra-coordinator-jar:5.10.5.Final
 org.jboss.narayana.rts:lra-coordinator-jar:5.10.5.Final
 org.jboss.narayana.rts:lra-proxy-api:5.10.5.Final
+org.jboss.narayana.rts:lra-proxy-api:5.10.5.Final
 org.jboss.narayana.rts:lra-service-base:5.10.5.Final
+org.jboss.narayana.rts:lra-service-base:5.10.5.Final
+org.jboss.narayana.rts:narayana-lra:5.10.5.Final
 org.jboss.narayana.rts:narayana-lra:5.10.5.Final
```
There seem to be a number of duplicates here?

tevans78 replied:
At some point we should make the script that enforces the alphabetical order also check for duplicate lines

skasund commented:
L2 message review not required

idlewis commented:
#build

idlewis commented:
@tevans78 I rebased on the integration to pick up the changes to restructure the microprofile APIs.
At the same time, I've added the change to remove the duplicate entries in the oss_dependencies file

idlewis commented:
#build

idlewis commented:
#build

idlewis commented:
#build

idlewis commented:
#build

idlewis commented:
#build

idlewis commented:
#run-libby-bot

idlewis commented:
Merging on the basis of this build:
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_vWHl8PHREeq2j8eW2p4naw&tab=com.ibm.team.build.web.ui.internal.editors.result.link.LinksProvider
Which has no test failures which look relevant. The delivery here should be standalone and shouldn't affect any other components
